agent-inject/agent/annotations.go

```diff
@@ -394,3 +428,36 @@ func (a *Agent) preserveSecretCase(secretName string) (bool, error) {
 	}
 	return strconv.ParseBool(raw)
 }
+
+func (a *Agent) runAsSameID(pod *corev1.Pod) (bool, error) {
+	raw, ok := a.Annotations[AnnotationAgentRunAsSameUser]
+	if !ok {
+		return DefaultAgentRunAsSameUser, nil
+	}
+	runAsSameID, err := strconv.ParseBool(raw)
+	if err != nil {
+		return DefaultAgentRunAsSameUser, err
+	}
+	if runAsSameID {
+		if len(pod.Spec.Containers) == 0 {
```

Comment:
Probably OK to have it but can this ever be true?

Reply:
Yeah that's just to protect the next line, since it's referencing `[0]` directly.